gpytorch/kernels/rbf_kernel_grad.py

```diff
@@ -0,0 +1,152 @@
+        :attr:`active_dims` (tuple of ints, optional):
+            Set this if you want to compute the covariance of only a few input dimensions. The ints
+            corresponds to the indices of the dimensions. Default: `None`.
+        :attr:`lengthscale_prior` (Prior, optional):
+            Set this if you want to apply a prior to the lengthscale parameter.  Default: `None`.
+        :attr:`param_transform` (function, optional):
+            Set this if you want to use something other than softplus to ensure positiveness of parameters.
+        :attr:`inv_param_transform` (function, optional):
+            Set this to allow setting parameters directly in transformed space and sampling from priors.
+            Automatically inferred for common transformations such as torch.exp or torch.nn.functional.softplus.
+        :attr:`eps` (float):
+            The minimum value that the lengthscale can take (prevents divide by zero errors). Default: `1e-6`.
+
+    Attributes:
+        :attr:`lengthscale` (Tensor):
+            The lengthscale parameter. Size/shape of parameter depends on the
+            :attr:`ard_num_dims` and :attr:`batch_size` arguments.
+
+    Example:
+        >>> x = torch.randn(10, 5)
+        >>> # Non-batch: Simple option
+        >>> covar_module = gpytorch.kernels.ScaleKernel(gpytorch.kernels.RBFKernelGrad())
+        >>> covar = covar_module(x)  # Output: LazyTensor of size (60 x 60), where 60 = n * (d + 1)
+        >>>
+        >>> batch_x = torch.randn(2, 10, 5)
+        >>> # Batch: Simple option
+        >>> covar_module = gpytorch.kernels.ScaleKernel(gpytorch.kernels.RBFKernelGrad())
+        >>> # Batch: different lengthscale for each batch
+        >>> covar_module = gpytorch.kernels.ScaleKernel(gpytorch.kernels.RBFKernelGrad(batch_size=2))
+        >>> covar = covar_module(x)  # Output: LazyTensor of size (2 x 60 x 60)
+    """
+
+    def __init__(
+        self,
+        batch_size=1,
+        active_dims=None,
+        lengthscale_prior=None,
+        param_transform=softplus,
+        inv_param_transform=None,
+        eps=1e-6,
+        **kwargs
+    ):
+        # TODO: Add support for ARD
+        super(RBFKernelGrad, self).__init__(
+            batch_size=batch_size,
+            active_dims=active_dims,
+            lengthscale_prior=lengthscale_prior,
+            param_transform=softplus,
+            inv_param_transform=inv_param_transform,
+            eps=eps,
+            **kwargs
+        )
+
+    def forward(self, x1, x2, diag=False, **params):
+        b = 1
+        if len(x1.size()) == 2:
+            n1, d = x1.size()
+            n2, d = x2.size()
+        else:
+            b, n1, d = x1.size()
+            _, n2, _ = x2.size()
+
+        K = torch.zeros(b, n1 * (d + 1), n2 * (d + 1), device=x1.device, dtype=x1.dtype)  # batch x n1(d+1) x n2(d+1)
+        ell = self.lengthscale.squeeze()
```

Comment:
This squeeze will cause broadcasting issues if `batch_size != 1`. Is it necessary?

Reply:
I get a shape error when I call `likelihood(model(test_x)).confidence_region()` if I remove this squeeze. Is there an obvious reason why?